conn.go

```diff
@@ -261,6 +268,13 @@ func Server(conn net.Conn, config *Config) (*Conn, error) {
 	return ServerWithContext(ctx, conn, config)
 }
 
+// PacketServer listens for incoming DTLS connections.
+// Unlike Server, PacketServer allows for connections to change remote address.
+// The provided rAddr will be used as the initial remote address for sending.
+func PacketServer(ctx context.Context, conn net.PacketConn, rAddr net.Addr, config *Config) (*Conn, error) {
```

Comment:
As an alternative to adding `PacketServer`, we could update the existing `Server` and `Client` to require passing a `net.PacketConn` rather than `net.Conn`. The reasoning for introducing `PacketServer` was to minimize breaking changes, but if we are cutting a `v3` one could make the case that requiring the user to wrap a `net.Conn` before passing it to the `Server` or `Client` would make it more obvious that remote address updates would not be supported. Said another way, if we are going to have breaking changes, it may be worth doing a larger refactor.

However, I also want to balance getting this functionality out to folks sooner rather than later, and I anticipate that we may have more breaking changes when we add support for DTLS 1.3.

Reply:
Changing to `net.PacketConn` is something we've wanted to do for a long time, as it's confusing in many places for people to have a `net.Conn` in the context of UDP. It came up in #367 and we tagged it as a 3.0.0 milestone. So I'd be all in favour for changing that, as that's something we want to do one way or another.

Reply:
@daenney thanks for this feedback! I'll push up an example of how the change could look to get your thoughts 👍🏻